[Use Cases] Add use cases for mounting (#4773)
ElektraInitiative/libelektra: doc/usecases/mount/UC_list_mountpoints.md

# Use Case: List all mountpoints

## Summary

- **Scope:** Mounting
- **Level:** Kite
- **Actors:** Administrators
- **Brief:** The administrators can query the current mountpoints.

> - **Status:** Implemented (currently only file-based data sources can be mounted)

Reviewer: ```suggestion
- **Status:** Draft (currently only file-based data sources can be listed)
```

## Scenarios

- **Precondition:**

  - The administrators have a working installation of Elektra and a tool to access the KDB (key database) on their systems.

- **Main success scenario:**

>   - The administrators use Elektra to query a list with all current mountpoints.
>   - Elektra returns the list with the mountpoints and the data source identifiers. (e.g. filename and path, data source name)
>   - The administrators mount a new data source. (see [Use Case: Mount a data source](./UC_mount.md))
>   - The administrators use Elektra to query a list with all current mountpoints.
>   - The new mountpoint is now added to the list.
>   - The administrators unmount the previously mounted data source. (see [Use Case: Unmount a data source](./UC_unmount.md))
>   - The administrators use Elektra to query a list with all current mountpoints.
>   - The mountpoint for the unmounted data source is now not included in the list.

Reviewer: The other use cases already cover mounting/umounting:

```suggestion
  - The administrators use Elektra to query a list of all current mountpoints.
```

- **Alternative scenario:**

  - none new
  - The scenarios of the used use cases (mounting and unmounting) apply.

- **Error scenario:**

>   - An error occurred while mounting the data source. (see [Use Case: Mount a data source](./UC_mount.md))
>     - Elektra reports the error to the administrators.
>   - An error occurred while unmounting the data source. (see [Use Case: Unmount a data source](./UC_unmount.md))
>     - Elektra reports the error to the administrators.

Reviewer: I don't think we need an error scenario here.

```suggestion
```
